exporter/metric/metric.go

```diff
@@ -499,7 +497,7 @@ func (me *metricExporter) recordToTypedValueAndTimestamp(r *export.Record) (*mon
 	case apimetric.CounterKind, apimetric.UpDownCounterKind, apimetric.SumObserverKind, apimetric.UpDownSumObserverKind:
 		// CUMULATIVE measurement should have the same start time and increasing end time.
 		// c.f. https://cloud.google.com/monitoring/api/ref_v3/rest/v3/projects.metricDescriptors#MetricKind
-		if sum, ok := agg.(*sum.Aggregator); ok {
+		if sum, ok := agg.(*array.Aggregator); ok {
```

Comment:
@liiling could you try adding a unit test reproducing this issue for both existing metric types (Counter) and new (SumObserver) to make sure we didn't break anything and fixed the issue?

Reply:
Also, I don't think this change is correct. Looking at https://github.com/open-telemetry/opentelemetry-go/blob/master/sdk/metric/selector/simple/simple.go it appears that the default aggregation for these types is still always `sum`.

The default aggregation for Value Observers can be configured, so we will need to rethink how we handle that in the future, but that is covered by the existing TODO comment

Reply:
Thanks for the review! I agree that the change seems strange... Will add test cases to reproduce the issue as suggested!

Reply:
Sorry for the delayed response.

Yea that's interesting. I hadn't noticed the custom selector we are using in this repo sorry. I'm not sure I fully understand the justification in the comments there as to why we are using array: "array gives the option to use Sum, Count, Max, Min, Quantile and Points (most flexible)". This doesn't seem to apply given that we always extract the sum here anyway.

I would probably recommend dropping that custom selector as I think we're better to follow what the SDK is doing and wait until a "views" or "default aggregations" implementation is implemented there before deciding what to do in this repo. If we drop that custom selector, we should be able to just change this code back to `sum` as per what the SDK's `simple.go` selector does.

In either case, at some point I think we should change this to switch on the aggregation type instead of the metric type, as per https://github.com/open-telemetry/opentelemetry-go/blob/master/exporters/metric/prometheus/prometheus.go#L223

@ymotongpoo what are your thoughts on this?

Reply:
Thank you for raising this to the discussion. The reason why I used `array` instead of `sum` is exactly because of what @james-bebbington pointed out. At the moment I implemented the selector.go, the predefined selectors didn't provide the option to fetch LastValue as noted in the comment here.
https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/blob/84f7c391be/exporter/metric/metric.go#L487-L489

I haven't checked the latest status of predefined selectors in SDK, but if they support the LastValue now, then we can remove the custom selector entirely and use the standard one instead. Otherwise, as James says, Views API or default aggregations for LastValue would be the alternative.

Reply:
Oh I see what you mean - looks like we do still need this selector for now to select LastValue 👍  (doesn't look like there is a selector in this SDK for do this)

Lets just change the default aggregator in `selector.go` to sum for now then as @liiling suggested

Reply:
@james-bebbington Thanks for the update! The suggested changes are now implemented.

Reply:
FYI I brought up the issue with all SDK selectors using histogram for `ValueObserver`, and the agreement is that we may upstream a new selector which will use last value instead.
Then we'll be able to remove custom selector from our repo.

There is also a similar issue about Prometheus using histogram for exporting gauges: https://github.com/open-telemetry/opentelemetry-go/issues/1139#issuecomment-690750369